db/import.go

```diff
@@ -155,7 +164,9 @@ func (db *Database) importDoc(docid string, body Body, isDelete bool, existingDo
 
 		// If the existing doc is a legacy SG write (_sync in body), check for migrate instead of import.
 		_, ok := body[base.SyncPropertyName]
-		if ok {
+		// Also check for sync data in existingDoc body as some routes end with sync data being extracted from body
```

Comment:
I thought the whole point of doc.inlineSyncData was to flag this condition (sync data being extracted from the body) at the time it happens.  Under what scenario are we calling importDoc, and we no longer reliably know the origin?

In general I don't think it's reasonable to rely on scanning for `"_sync":` in the bytes - both for the unnecessary performance hit, and also the potential for false positives.